### ISSUE TITLE
Display actual scanner name in 'Found By' column

### ISSUE THREAD

**Issue by rsenden** [DefectDojo/django-DefectDojo#3797]
When importing scan results, DefectDojo displays the importer name in the 'Found By' column. This is fine for specific importers, however for generic importers like SARIF or GitLab SAST it would be more useful to see the actual scanner name in the 'Found By' column (if available in the SARIF or GitLab report). 

Especially if you are using multiple tools that generate the same type of reports (i.e. multiple tools that generate a GitLab SAST report), just seeing `GitLab SAST Report` in the list of findings is not very useful. More useful would be to see the actual scanner names like `PMD`, `ESLint`, `Brakeman`, or `Fortify`.

![image](https://user-images.githubusercontent.com/8635138/107026847-9f9f0700-67ab-11eb-8fcb-f7658c4b7181.png)

**Comment by damiencarol** [comment]
@rsenden I totally agree. To do that we need to refactor some internal stuffs like how we manage results of parsers.
This is a work in progress.

**Comment by rsenden** [comment]
To prevent this issue from automatically closing, has there been any progress on this? 

Also I think the title change made by @valentijnscholten doesn't properly describe this issue; this issue doesn't apply to only SARIF import but also other types of imports like GitLab reports or generic JSON import.

For example, for generic JSON import you'd also want to be able to set and display the actual scanner name. However trying to set the `found_by` property in the JSON file results in an error message `An exception error occurred during the report import:Direct assignment to the forward side of a many-to-many set is prohibited. Use found_by.set() instead`.

**Comment by damiencarol** [comment]
@rsenden handling `found_by` in generic JSON need some work.

**Comment by damiencarol** [comment]
@rsenden started to work on this one. see #5121
